src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java

```diff
@@ -554,4 +562,10 @@ private String unexpectedTransactionState(InnReachTransaction transaction) {
     return UNEXPECTED_TRANSACTION_STATE + transaction.getState();
   }
 
+  private LocalAgency findLocalAgency(String code) {
+    var localAgency = localAgencyRepository.fetchOneByCode(code);
+    Assert.isTrue(localAgency.isPresent(), "Can't find local agency for code = " + code);
+    return localAgency.get();
```

Comment:
we usually use orElseThrow() with EntityNotFoundException like:
`.orElseThrow(() -> new EntityNotFoundException("Central server with code: " + code + " not found"));`